core/src/main/scala/com/yahoo/maha/core/query/oracle/OuterGroupByQueryGenerator.scala

```diff
@@ -0,0 +1,507 @@
+              whereFilters += f.filter
+            } else if (fact.factColMap.contains(name)) {
+              val column = fact.columnsByNameMap(name)
+              val alias = queryContext.factBestCandidate.factColMapping(name)
+              val exp = column match {
+                case FactCol(_, dt, cc, rollup, _, annotations, _) =>
+                  s"""${renderRollupExpression(name, rollup)}"""
+                case OracleDerFactCol(_, _, dt, cc, de, annotations, rollup, _) =>
+                  s"""${renderRollupExpression(de.render(name, Map.empty), rollup)}"""
+                case any =>
+                  throw new UnsupportedOperationException(s"Found non fact column : $any")
+              }
+              val f = FilterSql.renderFilter(
+                filter,
+                queryContext.factBestCandidate.publicFact.aliasToNameColumnMap,
+                fact.columnsByNameMap,
+                OracleEngine,
+                literalMapper,
+                Option(exp)
+              )
+              escaped |= f.escaped
+              havingFilters += f.filter
+            } else {
+              throw new IllegalArgumentException(
+                s"Unknown fact column: publicFact=${publicFact.name}, fact=${fact.name} alias=${filter.field}, name=$name")
+            }
+        }
+      }
+      val dayFilter = FilterSql.renderFilter(
+        requestModel.localTimeDayFilter,
+        queryContext.factBestCandidate.publicFact.aliasToNameColumnMap,
+        fact.columnsByNameMap,
+        OracleEngine,
+        literalMapper).filter
+
+      val combinedQueriedFilters = {
+        if (hasPartitioningScheme) {
+          val partitionFilters = new mutable.LinkedHashSet[String]
+          val partitionFilterOption = partitionColumnRenderer.renderFact(queryContext, literalMapper, OracleEngine)
+          if(partitionFilterOption.isDefined) {
+            partitionFilters += partitionFilterOption.get
+            AndFilter(partitionFilters ++ whereFilters)
+          } else {
+            AndFilter(whereFilters + dayFilter)
+          }
+        } else {
+          AndFilter(whereFilters + dayFilter)
+        }
+      }
+
+      val whereClauseExpression = s"""WHERE ${combinedQueriedFilters.toString} """
+      queryBuilder.setWhereClause(whereClauseExpression)
+
+      if (havingFilters.nonEmpty) {
+        val havingAndFilters = AndFilter(havingFilters.toSet)
+        val havingClauseExpression = s"""HAVING ${havingAndFilters.toString}"""
+        queryBuilder.setHavingClause(havingClauseExpression)
+      }
+
+    }
+
+    /*
+     Primitive columns are fact cols rendered in inner select which are dependent on the derived columns
+     Derived columns with dependencies are not rendered in ogb_generateFactViewColumns
+     Primitive columns are also rendered in preOuterSelect rendering
+      */
+    val primitiveColsSet = new mutable.LinkedHashSet[(Column, String)]()
+
+    def ogb_generateFactViewColumns(): Unit = {
```

Comment:
Can you name this without the "_" in the name?  E.g. ogbGenerateFactViewColumns or myGenerateFactViewColumns.

Also, can you make this private[this] since all these are not called outside of the generateDimFactOuterGroupByQuery context.